go/vt/vtgate/planbuilder/operators/ast_to_op.go

```diff
@@ -436,15 +436,8 @@ func createQueryTableForDML(
 func addColumnEquality(ctx *plancontext.PlanningContext, expr sqlparser.Expr) {
 	switch expr := expr.(type) {
 	case *sqlparser.ComparisonExpr:
-		if expr.Operator != sqlparser.EqualOp {
-			return
-		}
-
-		if left, isCol := expr.Left.(*sqlparser.ColName); isCol {
-			ctx.SemTable.AddColumnEquality(left, expr.Right)
-		}
-		if right, isCol := expr.Right.(*sqlparser.ColName); isCol {
-			ctx.SemTable.AddColumnEquality(right, expr.Left)
+		if expr.Operator == sqlparser.EqualOp || expr.Operator == sqlparser.NullSafeEqualOp {
```

Comment:
This looks like a behavioural change to me where support for the `<=>` operator was added. I think this makes sense, but it would be good to either add tests that cover this (seems like we don't have any), or extract it to a separate PR if this is worthy of being backported. 🙇‍♂️

Reply:
removed via 524c7e34896